include/eve/arch/cpu/wide.hpp

```diff
@@ -121,59 +122,226 @@ namespace eve
     EVE_FORCEINLINE wide operator--(int)  noexcept  { auto that(*this); operator--(); return that; }
 
     //==============================================================================================
-    // Compound operators
+    // Bitwise operators
+    //==============================================================================================
+    friend EVE_FORCEINLINE auto operator~(wide const& v) noexcept
+    {
+      return detail::self_bitnot(v);
+    }
+
+    friend  EVE_FORCEINLINE auto operator&=(wide& w, value auto o) noexcept
+                        ->  decltype(detail::self_bitand(w, o))
+    {
+      return detail::self_bitand(w, o);
+    }
+
+    template<scalar_value U,typename M>
+    friend EVE_FORCEINLINE    auto operator&(wide const& v, wide<U,M> const& w) noexcept
+                          ->  std::decay_t<decltype(std::declval<wide&>() &= w)>
+    {
+      auto    that  = v;
+      return  that &= w;
+    }
+
+    friend EVE_FORCEINLINE auto operator&(wide const& v, scalar_value auto w) noexcept
+    requires (sizeof(w) == sizeof(Type))
+    {
+      auto    that  = v;
+      return  that &= bit_cast(w, as_<Type>{});
```

Comment:
Why are you allowing to and with anything of the same size? That's a bit loose.

Reply:
That's what we do since version 0. We want BITwise operations to works on BIT, no matter the underlying type as what is important here is that each bit has a counterpart.

It helps writing a lot of SIMD math where you do bit and/or/wor between floating point values and masks.